File: Online-store-API/src/main/java/com/ijse/lk/OnlinestoreAPI/dto/ProductDTO.java
package com.ijse.lk.OnlinestoreAPI.dto;

import com.ijse.lk.OnlinestoreAPI.entity.ProductCategory;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Date;

@Data // getters & setters
@AllArgsConstructor
@NoArgsConstructor
@ToString

public class ProductDTO {
    private String id;
//    private ProductCategory category;
    private String category;
    private String name;
    private String description;
    private double unit_price;
    private String image_url;
    private boolean active;
    private int units_in_stock;
    private Date date_created;
    private Date last_updated;
}

File: OnlineStoreApp/src/app/dto/ProductDTO.ts
export class ProductDTO {
    public constructor(
        public id: string,
        public category: string,
        public name: string,
        public description: string,
        public unit_price: string,
        public image_url: string,
        // public active: string,
        public units_in_stock: string
        // public date_created: string,
        // public last_updated: string
    ){
    }
}